cms/djangoapps/contentstore/management/commands/migrate_cert_config.py

```diff
@@ -0,0 +1,242 @@
+        certificate_config = create_course_certificate_config_data(str(course_key), mode.slug, certificate_data)
+        files_to_upload = dict(get_certificate_signature_assets(certificate_config))
+        certificate_config_data = attr.asdict(certificate_config)
+        send_course_certificate_configuration(str(course_key), certificate_config_data, files_to_upload)
+
+    def delete_from_store(self, course: CourseBlock, certificates: List[Dict[str, str]]):
+        """
+        Deletes certificate configuration from modulestore storage.
+        """
+        store = modulestore()
+        request = FakedRequest(ModuleStoreEnum.UserID.mgmt_command)
+        for cert in certificates:
+            CertificateManager.remove_certificate(
+                request=request, store=store, course=course, certificate_id=cert['id']
+            )
+
+    def validate_input(self, options: Dict[str, str]):
+        """
+        Makes manage-command input validation. Raises CommandError if has conflicts.
+        """
+        if (not len(options['course_ids']) and not options.get('course_storage_type')) or (
+            len(options['course_ids']) and options.get('course_storage_type')
+        ):
+            raise CommandError(
+                'Certificate configurations migration requires one or more <course_id>s '
+                'OR the --course_storage_type choice.'
+            )
+
+    def get_course_keys_by_option(self, options: Dict[str, str]) -> Iterator[CourseKey]:
+        storage_type = options['course_storage_type']
+        course_ids = options['course_ids']
+        if storage_type:
+            if storage_type == 'all':
+                return chain(self.get_mongo_courses(), self.get_split_courses())
+            elif storage_type == 'draft':
+                return self.get_mongo_courses()
+            elif storage_type == 'split':
+                return self.get_split_courses()
+        if course_ids:
+            return map(self._parse_course_key, course_ids)
+
+    def migrate(self, course_keys: List[CourseKey], options: Dict[str, str]):
+        """
+        Main entry point for executiong all migration-related actions.
+
+        Sending to Credential and/or removal from storage.
+        If there are problems with some course, i.e. or it does not exist, or the course is not set to mode,
+        which allows to have a certificate, or no certificate configuration,
+        then an user of this command will be notified by a message.
+        """
+        for course_key in course_keys:
+            if course := modulestore().get_course(course_key):
+                if course_modes := CourseMode.objects.filter(
+                    course_id=course_key, mode_slug__in=CourseMode.CERTIFICATE_RELEVANT_MODES
+                ):
+                    if certificates := CertificateManager.get_certificates(course):
+                        for certificate_data in certificates:
+                            for mode in course_modes:
```

Comment:
I understand correctly that several identical certificates with different mods can be sent to the Credentials here?

Reply:
Yes, each mode that can have a certificate will get a request to Cred.

Reply:
okay, resolved